diff --git a/macros.efi-srpm.in b/macros.efi-srpm.in
index 00503f1..9800447 100644
--- a/macros.efi-srpm.in
+++ b/macros.efi-srpm.in
@@ -29,21 +29,27 @@
   local function getarch()
     if arch("ia64") then
       return("ia64")
-    elseif arch("x86_64") then
+    elseif arch("%{x86_64}") then
       return("x64")
     elseif arch("%{ix86}") then
       return("ia32")
-    elseif arch("aarch64") then
+    elseif arch("%{aarch64}") then
       return("aa64")
+    elseif arch("%{riscv64}") then
+      return("rv64")
+    elseif arch("%{riscv32}") then
+      return("rv32")
     elseif arch("%{arm}") then
       return("arm")
+    elseif arch("%{loongarch64}") then
+      return("loongarch64")
     else
       return("none")
     end
   end
   -- alt arch
   local function getaltarch()
-    if arch("x86_64") then
+    if arch("%{x86_64}") then
       return("ia32")
     else
       return("none")
